onnxruntime/core/providers/cpu/tensor/col2im.cc

```diff
@@ -0,0 +1,141 @@
+    col_input_data[i] << ", "; std::cout << ") with shape "<< col_input_shape << std::endl;
+  std::cout << "\tInput 1: image_shape = ("; for (auto i=0; i < image_shape->Shape().Size(); ++i) std::cout <<
+    image_shape->Data<int64_t>()[i] << ", "; std::cout << ")" << std::endl;
+  std::cout << "\tInput 2: kernel_shape = ("; for (auto i=0; i < kernel_shape->Shape().Size(); ++i) std::cout <<
+    kernel_shape->Data<int64_t>()[i] << ", "; std::cout << ")" << std::endl;
+  std::cout << "\tAttribute strides = ("; for (size_t i=0; i < col2im_attrs_.strides.size(); ++i) std::cout <<
+    col2im_attrs_.strides[i] << ", "; std::cout << ")"<< std::endl;
+  std::cout << "\tAttribute dilations = ("; for (size_t i=0; i < col2im_attrs_.dilations.size(); ++i) std::cout <<
+    col2im_attrs_.dilations[i] << ", "; std::cout << ")"<< std::endl;
+  std::cout << "\tAttribute pads = ("; for (size_t i=0; i < col2im_attrs_.pads.size(); ++i) std::cout <<
+    col2im_attrs_.pads[i] << ", "; std::cout << ")"<< std::endl;
+
+  std::cout << "\tVariable C: " << C << std::endl;
+  std::cout << "\tVariable col_input_N = " << col_input_N << std::endl;
+  std::cout <<  "\tVariable image_shape_size: " << image_shape_size << std::endl;
+  std::cout <<  "\tVariable kernel_shape_size: " << kernel_shape_size << std::endl;
+
+  std::cout << "\n\tStatus Col2Im<T>::Compute() --> math::Col2imNd<>()" << std::endl;
+
+  assert(image_shape_size == Y_offset);  // just for temp debug
+
+  for (auto image_id = 0; image_id < col_input_N; ++image_id) {
+    std::cout << "Image " << image_id+1 << " out of "<< col_input_N << std::endl;
+    if (image_shape->Shape()[0] == 2) {
+      std::cout << "image_shape->Shape()[0] == 2 --> Col2Im" << std::endl;
+      math::Col2im<float, CPUMathUtil, StorageOrder::NCHW>(
+        col_input_data + image_id * col_output_stride,
+        C,
+        image_shape->Data<int64_t>()[0],
+        image_shape->Data<int64_t>()[1],
+        kernel_shape->Data<int64_t>()[0],
+        kernel_shape->Data<int64_t>()[1],
+        col2im_attrs_.dilations[0],
+        col2im_attrs_.dilations[1],
+        col2im_attrs_.pads[0],
+        col2im_attrs_.pads[1],
+        col2im_attrs_.pads[2],
+        col2im_attrs_.pads[3],
+        col2im_attrs_.strides[0],
+        col2im_attrs_.strides[1],
+        Ydata + image_id * col_input_stride,
+        &CPUMathUtil::Instance());
+    } else {
+      std::cout << "image_shape->Shape()[0] != 2 --> Col2ImNd (nd=" << image_shape->Shape()[0] << ") " << std::endl;
+      math::Col2imNd<T, CPUMathUtil, StorageOrder::NCHW>(
+        col_input_data + image_id * col_output_stride,    // const T* data_col,
+        image_shape->Data<int64_t>(),                     // const int64_t* img_shape,
+        Yshape.Slice(2).GetDims().data(),                 // const int64_t* output_shape,
+        // col_input_shape[1],                               // int64_t channels_col,
```

Comment:
@gramalingam I have tried `col_input_shape[1]` as suggested. The result is on comments. Tried `col_input_shape[2]` and it doesnt work as well